objects/client.c

```diff
@@ -2834,25 +2779,6 @@ luaA_client_get_icon(lua_State *L, client_t *c)
 }
 
 static int
-luaA_client_get_focusable(lua_State *L, client_t *c)
-{
-    bool ret;
-
-    if (c->focusable_set)
-        ret = c->focusable;
-
-    /* A client can be focused if it doesnt have the "nofocus" hint...*/
-    else if (!c->nofocus)
-        ret = true;
-    else
-        /* ...or if it knows the WM_TAKE_FOCUS protocol */
-        ret = client_hasproto(c, WM_TAKE_FOCUS);
```

Comment:
> WM_PROTOCOLS to Lua, but for that we'd need a good use case first; I don't think this one is one)

I would also like to move maximize, apply size_hints and fullscreen code to lua, but they also interact with some xprops. I missed this one, I should been more careful.. But given the large amount of boilerplate code in C. Could some kind of hidden `c:set_xproperty()` take care of that. Eventually, I could also like to move capi.tag to lua, but that again require being able to set a few workspace properties.

It also like to reduce the amount of code in the C core so I can share more with the eventual Wayland port. I know some concepts (such as this one!), wont match the behavior of the Wayland equivalent, but still.